### PR TITLE
 New NetworkManager PS Package

### PR DESCRIPTION
**Pull request by yanfa317**
## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by yanfa317** [comment]
Here is doc design https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1123

File: src/Network/Network/NetworkManager/NetworkGroup/RemoveAzNetworkGroupCommand.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            base.Execute();
+            ConfirmAction(
+                Force.IsPresent,
+                string.Format(Properties.Resources.RemovingResource, Name),
+                Properties.Resources.RemoveResourceMessage,
+                Name,
+                () =>
+                {
+                    this.NetworkGroupClient.Delete(this.ResourceGroupName, this.NetworkManagerName, this.Name);
+                    if (PassThru)
```
Comment (BethanyZhou) [inline src/Network/Network/NetworkManager/NetworkGroup/RemoveAzNetworkGroupCommand.cs:72]:
```suggestion
                    if (PassThru.IsPresent)
```

File: src/Network/Network/NetworkManager/NewAzNetworkManagerScopeCommand.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using MNM = Microsoft.Azure.Management.Network.Models;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "NetworkManagerScope"), OutputType(typeof(PSNetworkManagerScopes))]
+    public class NewPSNetworkManagerScopeCommand : NetworkManagerBaseCmdlet
```
Comment (BethanyZhou) [inline src/Network/Network/NetworkManager/NewAzNetworkManagerScopeCommand.cs:30]:
```suggestion
    public class NewAzNetworkManagerScopeCommand : NetworkManagerBaseCmdlet
```

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

File: src/Network/Network/Az.Network.psd1 [hunk 3]
```diff
@@ -597,21 +655,59 @@ PrivateData = @{
...
-    - ''Set-AzFirewallPolicy''
-* Added new to create new ''SqlSetting'' object for creating Azure Firewall Policy
-    - ''New-AzFirewallPolicySqlSetting''
-* Added new cmdlet to support query Load Balancer inbound nat rule port mapping lists for backend addresses
-    - ''Get-AzLoadBalancerBackendAddressInboundNatRulePortMapping''
-    - Also updated cmdlets to support inbound nat rule V2 configurations
-        - ''New-AzLoadBalancerInboundNatRuleConfig''
-        - ''Set-AzLoadBalancerInboundNatRuleConfig''
-        - ''Add-AzLoadBalancerInboundNatRuleConfig'''
+        ReleaseNotes = '* Onboard Azure Virtual Network Manager Cmdlets for preview
```
[the remark this answers is not in the document]
Comment (Queenpuyu) [reply src/Network/Network/Az.Network.psd1:658]: Hi

File: src/Network/Network/NetworkManager/DeployAzNetworkManagerCommit.cs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Management.Network.Models;
+using Microsoft.Rest.Azure;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.Network.Models.NetworkManager;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet("Deploy", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "NetworkManagerCommit", SupportsShouldProcess = true), OutputType(typeof(PSNetworkManagerCommit))]
```
[the remark this answers is not in the document]
Comment (yanfa317) [reply src/Network/Network/NetworkManager/DeployAzNetworkManagerCommit.cs:26]: It was added from last time https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1033

File: src/Network/Network/ChangeLog.md [hunk 5]
```diff
@@ -68,6 +113,60 @@
...
+    - `New-AzNetworkManagerSecurityUserConfiguration` 
+    - `Remove-AzNetworkManagerSecurityUserConfiguration` 
+    - `Set-AzNetworkManagerSecurityUserConfiguration` 
+    - `Get-AzNetworkManagerSecurityUserRuleCollection` 
+    - `New-AzNetworkManagerSecurityUserRuleCollection` 
+    - `Remove-AzNetworkManagerSecurityUserRuleCollection` 
+    - `Set-AzNetworkManagerSecurityUserRuleCollection` 
+    - `Get-AzNetworkManagerSecurityUserRule` 
+    - `New-AzNetworkManagerSecurityUserRule` 
+    - `Remove-AzNetworkManagerSecurityUserRule`
```
Comment (BethanyZhou) [inline src/Network/Network/ChangeLog.md:151]: Please move all these messages to upcoming release section.

File: src/Network/Network/ChangeLog.md [hunk 6]
```diff
@@ -19,11 +19,56 @@
 --->
 
 ## Upcoming Release
-* Added support for retreiving the state of packet capture even when the provisioning state of the packet capture was failure
-    - `Get-AzNetworkWatcherPacketCapture`
-* Added support for accepting Vnet, Subnet and NIC resources as the TargetResourceId for the following cdmlets
-    - `Set-AzNetworkWatcherFlowLog`
-    - `New-AzNetworkWatcherFlowLog`
```
Comment (BethanyZhou) [inline src/Network/Network/ChangeLog.md:26]: Please do not delete these messages.

**Review by BethanyZhou — CHANGES_REQUESTED** [review]
Please resolve   "Az.Network","Microsoft.Azure.Commands.Network.NewAzNetworkManagerScopeCommand","New-AzNetworkManagerScope","1","8100","New-AzNetworkManagerScope Does not support ShouldProcess but the cmdlet verb New indicates that it should.","Determine if the cmdlet should implement ShouldProcess and if so determine if it should implement Force / ShouldContinue"

**Review by BethanyZhou — APPROVED** [review]

**Pull request merged**